lib/model/message_list.dart

```diff
@@ -86,6 +87,30 @@ class MessageListView extends ChangeNotifier {
     notifyListeners();
   }
 
+  Message applyChangesToMessage(UpdateMessageEvent event, Message oldMessage) {
+    Map<String, dynamic> oldMessageJson = oldMessage.toJson();
+
+    //TODO: What other fields should be edited by the update event?
+    oldMessageJson["content"] = event.renderedContent;
+    oldMessageJson["lastEditTimestamp"] = event.editTimestamp;
+
+    return Message.fromJson(oldMessageJson);
+  }
+
+  void maybeUpdateMessage(UpdateMessageEvent event) {
+    int oldIdx = messages.indexWhere((m) => m.id == event.messageId);
```

Comment:
Let's use a binary search for this — the messages are sorted by message ID. There could be a few thousand messages in the list (once we have #78, which is coming soon), which should be enough to make that more efficient.

Reply:
👍 Will swap to Flutter's implementation of `binarySearch`